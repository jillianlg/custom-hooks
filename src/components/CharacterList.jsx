import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
// import { Link } from 'react-router-dom';

const CharacterList = ({ characters }) => {
  const characterElements = characters.map(character => (
    <li key={character.id}>
      {/* <Link to={`/characters/${character.id}`}> */}
        <Character image={character.image} name={character.name} />
      {/* </Link> */}
    </li>
  ));

  return (
    <ul data-testid="characters">
      {characterElements}
    </ul>
  );
};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })).isRequired
};

export default CharacterList;
