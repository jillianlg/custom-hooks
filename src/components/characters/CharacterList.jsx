/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
import { Link } from 'react-router-dom';
import styles from './Characters.css';
import { useTheme } from '../../state/themeModeContext';

const CharacterList = ({ characters }) => {
  const { themeMode } = useTheme();

  const characterElements = characters.map(character => (
    <div key={character.id}>
      <Link to={`/characters/${character.id}`}>
        <Character image={character.image} name={character.name} />
      </Link>
    </div>
  ));

  return (
    <div className={`${styles.characterList} ${styles[themeMode]}` } data-testid="characters">
      {characterElements}
    </div>
  );
};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })).isRequired
};

export default CharacterList;
