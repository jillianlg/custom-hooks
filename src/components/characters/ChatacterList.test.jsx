import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';
import { MemoryRouter } from 'react-router-dom';

describe('CharacterList component', () => {
  afterEach(() => cleanup());
  it('renders CharacterList', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterList 
          characters={[]} />
      </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});