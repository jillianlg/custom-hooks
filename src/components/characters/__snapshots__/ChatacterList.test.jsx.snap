// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<DocumentFragment>
  <div
    class="characterList"
    data-testid="characters"
  />
</DocumentFragment>
`;
